### ISSUE TITLE
no-unused-vars not working for assembly declared variables

### ISSUE THREAD

**Issue by fernandomg** [protofire/solhint#169]
Code:
```solidity
contract Foo {
  function h(address a) internal {
    assembly {
        let t := eq(a, and(mask, calldataload(4)))
    }
  }
}
```

Config:
```json
{
  "rules": {
    "no-unused-vars": "error"
  }
}
```

No error is emitted, while it should emit as `t` is not being used.
